src/core_plugins/kibana/public/dashboard/panel/dashboard_panel.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import { DashboardViewMode } from '../dashboard_view_mode';
+import { PanelHeader } from './panel_header';
+
+export class DashboardPanel extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {};
+    this.embeddable = null;
+    this.embeddableHandler = null;
+    this._isMounted = false;
+  }
+
+  async componentDidMount() {
+    this._isMounted = true;
+    const { getEmbeddableHandler, panel, getContainerApi } = this.props;
+
+    this.containerApi = getContainerApi();
+    this.embeddableHandler = getEmbeddableHandler(panel.type);
+    const editUrl = await this.embeddableHandler.getEditPath(panel.id);
```

Comment:
If we were to change this code to:

```
this.embeddableHandler.getEditPath(panel.id).then(editUrl => this.setState({ editUrl }));
this.embeddableHandler.getTitleFor(panel.id).then(title => this.setState({ title }));
```

it'd be roughly in-line with the first "not a warning" case outline [here](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-did-mount-set-state.md), with the sole difference being they're using call-backs instead of Promise. 

Even with Redux, we'll still be using actions to queue off async actions in ComponentDidMount, but the updated data will be coming through via our `props` as opposed to the state.